diff --git a/generate/efi/AcpiPkg_nostdlib.dsc b/generate/efi/AcpiPkg_nostdlib.dsc
index 99e83d1cf..c3625ef7b 100644
--- a/generate/efi/AcpiPkg_nostdlib.dsc
+++ b/generate/efi/AcpiPkg_nostdlib.dsc
@@ -43,4 +43,4 @@
   !endif  ## DEBUG_ENABLE_OUTPUT
 
 [Components.common]
-  AcpiPkg/source/acpidump.inf
+  efi/acpidump/acpidump_nostdlib.inf
diff --git a/generate/efi/acpidump/acpidump_nostdlib.inf b/generate/efi/acpidump/acpidump_nostdlib.inf
index 2558016f7..930155166 100644
--- a/generate/efi/acpidump/acpidump_nostdlib.inf
+++ b/generate/efi/acpidump/acpidump_nostdlib.inf
@@ -48,7 +48,7 @@
   components/utilities/utxferror.c
 
 [Packages]
-  AcpiPkg/AcpiPkg.dec
+  efi/AcpiPkg.dec
   MdePkg/MdePkg.dec
 
 [LibraryClasses]
diff --git a/source/components/utilities/utprint.c b/source/components/utilities/utprint.c
index 7fb55cea0..d88a04aee 100644
--- a/source/components/utilities/utprint.c
+++ b/source/components/utilities/utprint.c
@@ -565,7 +565,7 @@ vsnprintf (
     if (Size != ACPI_UINT32_MAX) {
         End = String + Size;
     } else {
-        End = ACPI_CAST_PTR(char, ACPI_UINT32_MAX);
+        End = ACPI_CAST_PTR(char, (UINT64)ACPI_UINT32_MAX);
     }
 
     for (; *Format; ++Format)
